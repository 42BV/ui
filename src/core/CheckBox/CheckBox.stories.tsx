import { storiesOf } from '@storybook/react';
import { CheckBox } from './CheckBox';
import React, { useState } from 'react';

storiesOf('Core/CheckBox', module)
  .addParameters({ component: CheckBox })
  .add('basic', () => {
    const [primary, setPrimary] = useState(false);

    return (
      <div className={'text-center'}>
        <CheckBox
          onChange={() => setPrimary(!primary)}
          checked={!primary}
          color={'primary'}
          ariaLabel={'primary'}
        />
      </div>
    );
  });
