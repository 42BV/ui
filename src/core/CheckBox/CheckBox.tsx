import React from 'react';
import { Changeable, UIBasePropsWithCSSPropertiesAndChildren } from '../types';

type CheckBoxProps = {
  checked?: boolean;

  onBlur?: React.FocusEventHandler;

  ariaLabel?: string;
} & Changeable<any, void> &
  Partial<
    Omit<
      UIBasePropsWithCSSPropertiesAndChildren<React.ReactNode>,
      'capture' | 'enterKeyHint' | 'form' | 'list'
    >
  >;

export function CheckBox({ ariaLabel, ...props }: CheckBoxProps) {
  return <input type={'checkbox'} aria-label={ariaLabel} {...props} />;
}
