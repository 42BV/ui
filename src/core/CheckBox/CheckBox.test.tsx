import React from 'react';
import { CheckBox } from './CheckBox';
import { render } from '@testing-library/react';

describe('Component: CheckBox', () => {
  function setup({
    id,
    checked = false,
    label,
    hiddenLabel
  }: {
    id: string;
    checked: boolean;
    label: string;
    hiddenLabel?: boolean;
  }) {
    const onChangeSpy = jest.fn();
    const onBlurSpy = jest.fn();

    const props = {
      id,
      onChange: onChangeSpy,
      checked,
      onBlur: onBlurSpy,
      ariaLabel: hiddenLabel && typeof label === 'string' ? label : undefined
    };

    const { container, rerender } = render(<CheckBox {...props} />);

    return { container, rerender, onChangeSpy, onBlurSpy };
  }

  describe('ui', () => {
    test('default', () => {
      const { container } = setup({
        id: '1',
        checked: false,
        label: 'Test',
        hiddenLabel: true
      });
      expect(container).toMatchSnapshot();
    });
  });
});
