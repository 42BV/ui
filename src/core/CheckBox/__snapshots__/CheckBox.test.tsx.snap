// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CheckBox ui default 1`] = `
<div>
  <input
    aria-label="Test"
    id="1"
    type="checkbox"
  />
</div>
`;
